[Chan Yu Cheng] IP

yucheng11122017 opened the pull request:
# Duke
> Follow your heart but take your brain with you. - Alfred Adder ([source](https://www.goodreads.com/quotes/623973-follow-your-heart-but-take-your-brain-with-you))

Your brain has limited space. 
Use Duke to remember all the things you need to do and never get scolded by your mother for not buying milk again 👍 ~~until you forget to add the task to Duke that is~~
``` Java
if (isDumb) {
    return useDuke;
}
```
Duke is
* Easy to use 
* ~~mostly~~ bug free
* Sacarstic so don't mess up!

Just 
1. Download the code
2. Run it 
3. Talk to Duke!
4. Get roasted by it!

Features: 
- [x] Supports different types of task
- [ ] Sends reminders to users

yucheng11122017 committed:
Add echo and exit when user types bye

yucheng11122017 committed:
'Level-2'

yucheng11122017 committed:
'Level-3'

yucheng11122017 committed:
Add todo, events, deadlines

yucheng11122017 committed:
Add automated testing

yucheng11122017 committed:
Add exception handling

yucheng11122017 committed:
Add delete

yucheng11122017 committed:
Remove unecessary code

yucheng11122017 committed:
Add task saving to hard disk

yucheng11122017 committed:
Add date formatting for deadline and event

yucheng11122017 committed:
Merge branch 'branch-Level-7'

yucheng11122017 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

yucheng11122017 committed:
Change format of saved tasks

yucheng11122017 committed:
Add UI, Filemanager, Parser, Tasklist, Command classes

yucheng11122017 committed:
Refactor code into packages

yucheng11122017 committed:
Add javadoc to public methods.

yucheng11122017 committed:
Add test for parser class - WIP.

yucheng11122017 committed:
Add junit test for parser and ui.

yucheng11122017 committed:
Change styling to follow convention.

yucheng11122017 committed:
Add Ui class.

yucheng11122017 committed:
Add feature to find tasks related to keyword.

yucheng11122017 committed:
Merge branch 'A-JavaDoc'

# Conflicts:
#	src/test/java/duke/ParserTest.java

yucheng11122017 committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	src/main/java/duke/FileManager.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/command/Command.java

yucheng11122017 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Tasklist.java
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/DeleteCommand.java

yucheng11122017 committed:
Add Gradle to run Duke.

yucheng11122017 committed:
Merge branch 'branch-A-Gradle'

yucheng11122017 committed:
Add checkstyle function.

yucheng11122017 committed:
Merge branch 'branch-A-checkstyle'

yucheng11122017 committed:
Add GUI

yucheng11122017 committed:
Merge branch 'branch-Level-10'

yucheng11122017 committed:
Find command input multiple arguments

yucheng11122017 committed:
Merge branch 'branch-A-Varargs'

yucheng11122017 committed:
Update Javadoc.

hungkhoaitay commented on build.gradle:
```diff
@@ -0,0 +1,70 @@
+plugins {
+    id 'java'
+    id 'application'
+    id 'checkstyle'
```
Nice, I didn't know about this before :)

hungkhoaitay commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,69 @@
+        if (splitBySpace.length < 2) {
+            throw new DukeException("Invalid input.");
+        }
+        if (splitBySpace[0].toLowerCase().equals("done")) {
```
You repeat this many times, perhaps can turns this into function? :)

hungkhoaitay reviewed (commented):
Overall, you did it well!

ramaven commented on gradlew:
```diff
@@ -0,0 +1,185 @@
+    i=0
+    for arg in "$@" ; do
+        CHECK=`echo "$arg"|egrep -c "$OURCYGPATTERN" -`
+        CHECK2=`echo "$arg"|egrep -c "^-"`                                 ### Determine if an option
```
Perhaps can add the comment in the line above rather than in the same line but far apart from the code

ramaven commented on src/main/java/duke/FileManager.java:
```diff
@@ -0,0 +1,80 @@
+                return tasks;
+            }
+        } catch (IOException e) {
+            System.out.println("Error occured initalising file");
```
Typo in "initalising" -> should be "initializing"

yucheng11122017 replied:
Right thanks!

ramaven commented on src/main/java/duke/command/CommandsTypes.java:
```diff
@@ -0,0 +1,13 @@
+/**
+ * Enumerates the types of commands that are supported by duke.
+ */
+public enum CommandsTypes {
```
According to the coding standards, the enums should be all uppercase

ramaven commented on src/main/java/duke/task/ToDos.java:
```diff
@@ -0,0 +1,40 @@
+/**
+ * Class that represents todo task.
+ */
+public class ToDos extends Task {
```
I think the class name could be "ToDo" or "ToDoTask" instead of "ToDos" as the "s" at the end may be a bit misleading as it may sound like it is referring to a list of todo tasks.

ramaven reviewed (commented):
Overall, your code is very well organised with clear class names and method names. I love the creativity in your version of Duke, and using your application seems like it would be very entertaining.

yucheng11122017 committed:
Update build gradle file

yucheng11122017 committed:
Merge branch 'master' of https://github.com/yucheng11122017/ip

yucheng11122017 committed:
Jar file is normal jar file

Users cannot run without dependencies

Let's add fat jar file with dependencies

yucheng11122017 committed:
Add assertions

Has no assertions in code

Hard to verfiy if code is working or where is bug located

Let's add assertions to make it easier to debug

yucheng11122017 committed:
Refactor code

Long methods present

Makes it hard to read and find bugs

Let's refactor code according to SLAP principles

yucheng11122017 committed:
Merge pull request #1 from yucheng11122017/branch-A-Assertions

Add assertions

yucheng11122017 committed:
Merge branch 'master' into branch-A-CodeQuality

yucheng11122017 committed:
Merge pull request #2 from yucheng11122017/branch-A-CodeQuality

Refactor code

yucheng11122017 committed:
Merge conflict

yucheng11122017 committed:
Merge branch 'master' of https://github.com/yucheng11122017/ip

# Conflicts:
#	src/main/java/duke/Parser.java

yucheng11122017 committed:
Fix bug where Delete command will execute markDone Command instead

yucheng11122017 committed:
Add gradle.yml

yucheng11122017 committed:
Add tagging

Users cannot add additional information to tasks

This makes it hard for them to remember these information

Let's add tagging so that they can group tasks together and add additional information.

yucheng11122017 committed:
Merge branch 'C-Tagging'

yucheng11122017 committed:
Add help

Did not have help before. Users have to figure out how to use on their own.

Let's add user so that user can have general help and also help on specific commands

Add help command so that user can get help.

yucheng11122017 committed:
Add User Guide for Duke.

yucheng11122017 committed:
Update formatting in User Guide Table

yucheng11122017 committed:
Update Ui with new profile pictures

yucheng11122017 committed:
Change code style.

yucheng11122017 committed:
Add jar file for release.

yucheng11122017 committed:
Update readme with link for jar file.

yucheng11122017 committed:
Remove debugging line.
